src/peep/peep.c

```diff
@@ -7772,13 +7772,385 @@ static void peep_on_enter_or_exit_ride(rct_peep *peep, int rideIndex, int flags)
 	}
 }
 
-/**
+/** Main logic to decide whether a peep should buy an item in question
+ *
+ * Seems to also handle the actual purchase as well, so once it returns, the
+ * peep will have the item and the money will have been deducted.
+ *
+ * eax: shopItem | (rideIndex << 8)
+ * ecx: price
+ * esi: peep*
+ *
+ * Returns 0 or 1 depending on if the peep decided to buy the item
  *
  *  rct2: 0x0069AF1E
  */
 static bool sub_69AF1E(rct_peep *peep, int rideIndex, int shopItem, money32 price)
 {
-	return !(RCT2_CALLPROC_X(0x0069AF1E, shopItem | (rideIndex << 8), 0, price, 0, (int)peep, 0, 0) & 0x100);
+#define STANDARD_FOOD_MASK 0x036BA3E0
+#define EXTRA_FOOD_MASK 0x0011FF78
+
+	rct_ride* ride = GET_RIDE(rideIndex);
+	sint16 value;
+
+	///sub_69AF1E
+	bool has_voucher;
+	RCT2_GLOBAL(0x00F1AEE3, uint8) = 0; // Seems to be a local bool for whether the peep has a matching voucher for this shop
```

Comment:
I had a look to see if 0x00F1AEE3 is used in anything else and it isn't so we can remove this.